src/Pos/Block/Logic.hs

```diff
@@ -828,12 +829,15 @@ createMainBlockFinish slotId pSk prevHeader = do
         let convertTx (txId, (tx, _, _)) = WithHash tx txId
         sortedTxs <- maybe onBrokenTopo pure $ topsortTxs convertTx localTxs
         sk <- Ether.asks' npSecretKey
-        -- for now let's be cautious and not generate blocks that are larger than
-        -- maxBlockSize/4
-        sizeLimit <- fromIntegral . toBytes . (`div` 4) <$> UDB.getMaxBlockSize
+        -- 50 bytes is substracted to account for different unexpected
+        -- overhead.  You can see that in bitcoin blocks are 1-2kB less
+        -- than limit. So i guess it's fine in general.
+        sizeLimit <-
+            fromIntegral . toBytes . (\x -> bool 0 (x - 100) (x > 100)) <$> UDB.getMaxBlockSize
         block <- createMainBlockPure
-                     sizeLimit prevHeader sortedTxs pSk
-                     slotId localPSKs sscData usPayload sk
+            sizeLimit prevHeader (map snd sortedTxs) pSk
+            slotId localPSKs sscData usPayload sk
+        lift $ logInfo $ "Created main block of size: " <> show (length $ Bi.encode block)
```

Comment:
Better use `biSize`, at least because with `store` it will most likely be more efficient than `length . encode`.